giza_datasets/loaders.py

```diff
@@ -89,4 +100,8 @@ def load(self, dataset_name):
                     "No .parquet files were found in the directory or subdirectories."
                 )
             df = self._load_multiple_parquet_files(parquet_files)
+        
+        if self.use_cache:
+            cache_manager_to_use = specific_cache_manager if specific_cache_manager is not None else self.cache_manager
+            cache_manager_to_use.save_to_cache(df, dataset_name)
```

Comment:
It seems that the flow is the following for a new dataset (first load):

* Load the parquet in polars
* If using the cache, then save it using polars to the cache

Other packages like `scikit` or `tensorflow` directly downloads the files to the cache and then reads it using the dependency that they want so a flow could be like this:

* If using the cache, download the files of the dataset (directory even), otherwise directly load from GCS
* Load local cache using polars

This way we could save time on polars writing to disk and make sure that we have the same files as in the bucket.

WDYT?

Reply:
It makes sense. All the logic is set up to do it the other way. As it doesn't affect the functionality, let's merge this version as is and create an issue to improve it in the next one.

Reply:
Finally I have included it in this PR